static/js/lightbox.js

```diff
@@ -66,27 +80,80 @@ export class PanZoomControl {
         });
     }
 
+    constrainImage(e) {
+        // Instead of using panzoom's built in bounds option which was buggy
+        // at the time of this writing, we act on pan/zoom events and move the
+        // image back in to view if it is moved beyond the image-preview container.
+
+        const {scale, x, y} = e.getTransform();
+        const image_width = $(".zoom-element > img")[0].clientWidth * scale;
+        const image_height = $(".zoom-element > img")[0].clientHeight * scale;
+        const zoom_element_width = $(".zoom-element")[0].clientWidth * scale;
+        const zoom_element_height = $(".zoom-element")[0].clientHeight * scale;
+        const max_translate_x = $(".image-preview")[0].clientWidth;
+        const max_translate_y = $(".image-preview")[0].clientHeight;
+
+        // Adjust this to control the number of pixels of the image
+        // that should always be visible.
+        const return_buffer = 50 * scale;
+
+        const zoom_border_width = (zoom_element_width - image_width) / 2 + image_width;
+        const zoom_border_height = (zoom_element_height - image_height) / 2 + image_height;
+        const modified_x = x + zoom_border_width;
+        const modified_y = y + zoom_border_height;
+
+        if (modified_x < 0) {
+            // Image has been dragged beyond the LEFT of the view.
+            const move_by = modified_x * -1;
+            e.moveBy(move_by + return_buffer, 0, false);
+        } else if (modified_x - image_width > max_translate_x) {
+            // Image has been dragged beyond the RIGHT of the view.
+            const move_by = modified_x - max_translate_x - image_width;
+            e.moveBy(-move_by - return_buffer, 0, false);
+        }
+
+        if (modified_y < 0) {
+            // Image has been dragged beyond the TOP of the view.
+            const move_by = modified_y * -1;
+            e.moveBy(0, move_by + return_buffer, false);
+        } else if (modified_y - image_height > max_translate_y) {
+            // Image has been dragged beyond the BOTTOM of the view.
+            const move_by = modified_y - max_translate_y - image_height;
+            e.moveBy(0, -move_by - return_buffer, false);
+        }
+    }
+
     reset() {
-        this.panzoom.reset();
+        this.panzoom.zoomAbs(0, 0, 1);
+        this.panzoom.moveTo(0, 0);
+        $("#lightbox_overlay").data("noclose", false);
```

Comment:
Maybe worth adding a comment to help understand why these are the right things to do to reset?

Reply:
Good idea, I've commented each of these steps.